### PR TITLE
test: Code Coverage: Personal Cards Page #3

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 89d1797</samp>

Added tests for personal card transactions feature in `PersonalCardsPage` component. The tests verify the functionality and UI of fetching, selecting, filtering, searching, and viewing transactions from the user's personal cards. The tests are written in `src/app/fyle/personal-cards/personal-cards.page.spec.ts`.

### <samp>🤖 Generated by Copilot at 89d1797</samp>

> _`PersonalCardsPage` is the portal to your doom_
> _You can't escape the transactions that consume_
> _Your soul is filtered, searched, and selected_
> _By the methods that you tested and neglected_

### <samp>🤖 Generated by Copilot at 89d1797</samp>

*  Add and remove imports for modules and components used in new features and tests ([link](https://github.com/fylein/fyle-mobile-app/pull/2402/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7L4-R17), [link](https://github.com/fylein/fyle-mobile-app/pull/2402/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7L22-R29))
*  Add tests for `loadData` and `segmentChanged` methods of `PersonalCardsPage` component, which fetch and filter personal card transactions based on page number and query parameters ([link](https://github.com/fylein/fyle-mobile-app/pull/2402/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7R323-R361))
*  Add tests for various methods of `PersonalCardsPage` component that handle user interactions and logic related to selecting, filtering, searching, and viewing personal card transactions, such as `selectExpense`, `onSelectAll`, `setState`, `onSearchBarFocus`, `onSimpleSearchCancel`, `onFilterPillsClearAll`, `onFilterClick`, `clearFilters`, `searchClick`, and `onFilterClose` ([link](https://github.com/fylein/fyle-mobile-app/pull/2402/files?diff=unified&w=0#diff-6e314ff0829ddfea53cbbca077b821775702ee8b5c5804cedab82e27a8a6f8a7R478-R610))

## Clickup
https://app.clickup.com/t/85ztx3nye

## Code Coverage
`59.53% Statements 178/299`
`42.85% Branches 21/49`
`60.27% Functions 44/73`
`59.79% Lines 177/296`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts [hunk 1]
```diff
@@ -433,6 +475,139 @@ describe('PersonalCardsPage', () => {
...
+    describe('selectExpense():', () => {
+      it('should add an expense to list', () => {
+        component.selectedElements = [];
+        fixture.detectChanges();
+
+        component.selectExpense('btxnMy43OZokde');
+        expect(component.selectedElements).toEqual(['btxnMy43OZokde']);
+      });
+
+      it('should add an expense to list', () => {
```
Comment (Chethan-Fyle) [inline src/app/fyle/personal-cards/personal-cards.page.spec.ts:488]: Description and actual action are different.

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts [hunk 2]
```diff
@@ -433,6 +475,139 @@ describe('PersonalCardsPage', () => {
...
+        component.selectedElements = ['btxnMy43OZokde'];
+        fixture.detectChanges();
+
+        component.selectExpense('btxnMy43OZokde');
```
Comment (Chethan-Fyle) [inline src/app/fyle/personal-cards/personal-cards.page.spec.ts:492]: Why is `selectExpense()` unselecting an expense? Does the function toggle between select and unselect? If yes, then the function name is misleading.

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts [hunk 3]
```diff
@@ -433,6 +475,139 @@ describe('PersonalCardsPage', () => {
...
+
+      component.onFilterPillsClearAll();
+
+      expect(component.clearFilters).toHaveBeenCalledTimes(1);
+    });
+
+    it('onFilterClick(): should open filter', fakeAsync(() => {
+      spyOn(component, 'openFilters');
+
+      component.onFilterClick('filters');
```
Comment (Chethan-Fyle) [inline src/app/fyle/personal-cards/personal-cards.page.spec.ts:543]: What is the use of passing a static value (`filters`) when the function name itself suggests we are clicking on filter?

**Review by Chethan-Fyle — APPROVED** [review]
minor comments and doubts.

**Review by OmkarJ13 — APPROVED** [review]

**Review by Dimple16 — COMMENTED** [review]
PR is reviewed, will skip

File: src/app/fyle/personal-cards/personal-cards.page.spec.ts (hunk 2 above)
Comment (jayfyle) [reply src/app/fyle/personal-cards/personal-cards.page.spec.ts:526]: Yes the function does unselect an expense, hence changing the name to `toggleExpense`

**Pull request merged**